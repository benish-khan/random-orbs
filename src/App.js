import React, { useState, useEffect} from 'react';
import * as txtgen from 'txtgen';
import { Button } from 'react-bootstrap';

import logo from './logo.svg';
import './App.css';

function App() {
  const [ randomText, setRandomText ] = useState('');

  const handleTextChange = () =>{
    const _new = txtgen.sentence();
    if(_new !== randomText)
      setRandomText(_new);
  }

  useEffect(()=>{
    setRandomText(txtgen.sentence());
  },[]);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>{randomText}</p>
        <Button name="Generate Text" onClick={handleTextChange}>Generate Text</Button>
      </header>
    </div>
  );
}

export default App;
