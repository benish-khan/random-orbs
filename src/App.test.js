import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme';
import App from './App';

describe('App', () => {

  it('renders generate text button', () => {
    const { getByText } = render(<App />);
    const buttonElement = getByText(/Generate Text/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it('Button Click Generates new Text', () => {
    const wrapper = mount(<App />);
    const _prev = wrapper.find('p').text();
    wrapper.find("button").simulate("click");
    expect(wrapper.find('p').text()).not.toEqual(_prev);
  });

});