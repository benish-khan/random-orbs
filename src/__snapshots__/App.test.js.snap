// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Button Click Generates new Text 1`] = `ReactWrapper {}`;
